src/Extensions/ExtensionManifest.re

```diff
@@ -6,76 +6,96 @@
-  [@deriving (show, yojson({strict: false, exn: true}))]
-  type t =
-    | [@name "ui"] Ui
-    | [@name "workspace"] Workspace;
-};
+module Json = Oni_Core.Json;
 
-module Author = {
-  [@deriving show]
-  type t = option(string);
+[@deriving show]
+type t = {
+  name: string,
+  version: string,
+  author: string,
+  displayName: option(LocalizedToken.t),
+  description: option(string),
+  // publisher: option(string),
+  main: option(string),
+  icon: option(string),
+  categories: list(string),
+  keywords: list(string),
+  engines: string,
+  activationEvents: list(string),
+  extensionDependencies: list(string),
+  extensionPack: list(string),
+  extensionKind: kind,
+  contributes: ExtensionContributions.t,
+  enableProposedApi: bool,
+}
 
-  let of_yojson_exn =
-    fun
-    | `String(v) => Some(v)
-    | v =>
-      Yojson.Safe.(
-        {
-          Util.member("name", v) |> Util.to_string_option;
-        }
-      );
+and kind =
+  | Ui
+  | Workspace;
 
-  let of_yojson = json =>
-    ResultEx.guard(() => of_yojson_exn(json))
-    |> Result.map_error(_ => "Error parsing author");
+module Decode = {
+  open Json.Decode;
 
-  let to_yojson = _author => `Null;
-};
+  let kind =
+    string |> map(fun
+    | "ui" => Ui
+    | "workspace" => Workspace
+    | _ => Ui
+    );
+
+  let author =
+    one_of([
+      ("string", string),
+      ("object", field("name", string)),
+    ]);
 
-module Engine = {
-  [@deriving (show, yojson({strict: false, exn: true}))]
-  type t = {vscode: string};
+  let engine =
+    field("vscode", string);
+
+  let manifest =
+    field("name", string) >>= name =>
+    field("version", string) >>= version =>
+    one_of([
+      ("author", field("author", author)),
+      ("publisher", field("publisher", string)),
+      ("default", succeed("Unknown Author")),
+    ]) >>= author =>
+    field_opt("displayName", LocalizedToken.decode) >>= displayName =>
+    field_opt("description", string) >>= description =>
+    field_opt("main", string) >>= main =>
+    field_opt("icon", string) >>= icon =>
+    field_opt("categories", list(string)) |> default([]) >>= categories =>
+    field_opt("keywords", list(string)) |> default([]) >>= keywords =>
+    field("engines", engine) >>= engines =>
+    field_opt("activationEvents", list(string)) |> default([]) >>= activationEvents =>
+    field_opt("extensionDependencies", list(string)) |> default([]) >>= extensionDependencies =>
+    field_opt("extensionPack", list(string)) |> default([]) >>= extensionPack =>
+    field_opt("extensionKind", kind) |> default(Ui) >>= extensionKind =>
+    field("contributes", ExtensionContributions.decode) >>= contributes =>
+    field_opt("enableProposedApi", bool) |> default(false) >>= enableProposedApi =>
+    succeed({
+      name, version, author, displayName, description, main, icon,
+      categories, keywords, engines, activationEvents, extensionDependencies,
+      extensionPack, extensionKind, contributes, enableProposedApi
+    });
```

Comment:
Significantly less readable for large decoders.

Note that there's also a "pipeline" API which is more readable, and can reduce the boilerplate a bit with generated record constructors, but it depends on the position of arguments and is therefore more error-prone, and also yields harder-to understand errors messages.